### ISSUE TITLE
[Bug][Numpy] The data type of power is not correct

### ISSUE THREAD

**Issue by sxjscience** [apache/incubator-mxnet#16653]
```python
import mxnet as mx
mx.npx.set_np()
a = mx.np.array(2, dtype=mx.np.int32)
b = a ** 3.1
print(b)
```
Returns
```
8
```

In the numpy the result should be:
```
import numpy as np
print(np.array(2) ** 3.1)
```
Returns
```
8.574187700290345
```

**Comment by xidulu** [comment]
I believe  https://github.com/apache/incubator-mxnet/blob/master/src/operator/numpy/np_elemwise_broadcast_op.cc#L32  is the root cause of this problem.
Also, `power` is not the only operator suffering from this bug :
```
>>> from mxnet import np, npx, autograd
>>> npx.set_np()
>>> a = np.array(2, dtype=np.int32)
>>> a
array(2, dtype=int32)
>>> a + 1.1
array(3, dtype=int32)
```
The `mixed precision` mechanism introduced in  https://github.com/apache/incubator-mxnet/pull/16631 may be a solution to this issue. @haojin2

**Comment by haojin2** [comment]
@xidulu yes, the mixed precision binary ops is the effort that aims to solve this kind of problem. Right now we have multiplication between `bool` and all other types, later we'll expand this support to more combinations. @sxjscience

**Comment by reminisce** [comment]
This is a known issue of many operators from which floats are expected with integral inputs. Instead of tweaking many `TYPE_SWITCH` macros to make it work, we plan to solve it using TVM to generate appropriate kernels for integral dtypes.

**Comment by samskalicky** [comment]
@lanking520 assign @reminisce

**Comment by haojin2** [comment]
The latest master branch today has this behavior:
```
>>> import mxnet as mx
>>> mx.npx.set_np()
>>> a = mx.np.array(2, dtype=mx.np.int32)
>>> b = a ** mx.np.array(3.1, dtype=mx.np.float32)
>>> b
array(8.574187)
```
But still has a little bit problem with the scalar case:
```
>>> import mxnet as mx
>>> mx.npx.set_np()
>>> a = mx.np.array(2, dtype=mx.np.int32)
>>> b = a ** 3.1
>>> print(b)
8
```
I think this is a new case for the binary scalar op, I'll work on a solution to this.

**Comment by yzhliu** [comment]
@Tommliu is working on this.

**Comment by yzhliu** [comment]
closed by #18277
